con-master/src/main/java/ba/codecentric/master/controller/FacilityController.java

```diff
@@ -27,7 +32,43 @@ public FacilityController(FacilityService facilityService, ImageService imageSer
 
     @PostMapping(value = "/facilities", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
     public Facility saveFacility(@RequestPart("image") MultipartFile image, @Valid @RequestPart("facility") Facility facility) throws IOException {
-        facility.setImage(imageService.saveImage(image.getInputStream(), image.getOriginalFilename()));
-        return facilityService.saveFacility(facility);
+        String imageName = "";
+        Integer facilityId = facility.getId();
+
+        if (facilityId != null) {
+            imageName = facilityService.findImagePathById(facilityId);
+            if (imageService.doesImageExist(imageName)) {
+                imageService.deleteImage(imageName);
+            }
+        }
+
+        String fileName = imageService.saveImage(image.getInputStream(), image.getOriginalFilename());
+        if (fileName != null) {
+            facility.setImage(fileName);
+            return facilityService.saveFacility(facility);
+        }
+        return new Facility();
+
+    }
+
+    @GetMapping(value = "/facilities")
+    public List<Facility> getAllFacility() {
+        return facilityService.getAllFacility();
+    }
+
+    @DeleteMapping(value = "/facilities/{facilityId}")
+    public void deleteFacility(@PathVariable Integer facilityId) throws Exception {
+
+        Facility facility = facilityService.findById(facilityId);
+
+        if (facility != null) {
```

Comment:
In case if facility is not found, we should return appropriate message and status code.